attrs.go

```diff
@@ -86,5 +99,18 @@ func fileStatFromInfo(fi os.FileInfo) (uint32, *FileStat) {
 	// os specific file stat decoding
 	fileStatFromInfoOs(fi, &flags, fileStat)
 
+	// if fi implements FileInfoUidGid, retrieve Uid, Gid from it
+	if fiExt, ok := fi.(FileInfoUidGid); ok {
+		flags |= sshFileXferAttrUIDGID
+		fileStat.UID = fiExt.Uid()
+		fileStat.GID = fiExt.Gid()
+	}
+
+	// if fi implements FileInfoExtendedData, retrieve extended data from it
+	if fiExt, ok := fi.(FileInfoExtendedData); ok {
+		flags |= sshFileXferAttrExtended
+		fileStat.Extended = fiExt.Extended()
```

Comment:
Consider that if `len(fileStat.Extended) == 0` we have a better encoding by not asserting the `sshFileXferAttrExtended` flag in the first place?

Reply:
True, this should obviously be checked.